runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -1240,6 +1240,16 @@ J9::SymbolReferenceTable::isFieldClassObject(TR::SymbolReference *symRef)
    return false;
    }
 
+bool
+J9::SymbolReferenceTable::isFieldTypeBool(TR::SymbolReference *symRef)
+   {
+   int32_t len;
+   const char *fieldName = symRef->getOwningMethod(comp())->fieldSignatureChars(symRef->getCPIndex(), len);
+   dumpOptDetails(comp(), "got field signature as %s\n", fieldName);
+   if (fieldName[len-2] != '[' && fieldName[len-1] == 'Z')
+      return true;
+   return false;
```

Comment:
Why not `return fieldName[len-2] != '[' && fieldName[len-1] == 'Z'` instead of these 3 lines?

Reply:
modified according to the comment. Thanks!